src/AutoRest.CSharp/Common/AutoRest/Plugins/Configuration.cs

```diff
@@ -596,5 +599,57 @@ private static bool ReadOption(JsonElement root, string option)
+        (
+            AbsoluteProjectFolder: AbsoluteProjectFolder,
+            AzureArm: AzureArm,
+            IsMgmtTestProject: MgmtTestConfiguration is not null,
+            LibraryName: LibraryName,
+            Namespace: Namespace,
+            SkipCSProjPackageReference: SkipCSProjPackageReference,
+            RelativeProjectFolder: RelativeProjectFolder,
+            Generation1ConvenienceClient: Generation1ConvenienceClient
+        );
+
+        // load CSharpProj configuration from Autorest input
+        internal static CSharpProjConfiguration InitializeCSharpProjConfig(IPluginCommunication autoRest, string defaultLibraryName, string defaultNamespace)
+        {
+            string outputFolder = GetOutputFolderOption(autoRest);
+            var projectFolder = GetProjectFolderOption(autoRest) ?? ProjectFolderDefault;
+            (var absoluteProjectFolder, var relativeProjectFolder) = ParseProjectFolders(outputFolder, projectFolder);
+
+            return new CSharpProjConfiguration(
+                AbsoluteProjectFolder: absoluteProjectFolder,
+                AzureArm: GetAzureArmOption(autoRest),
+                IsMgmtTestProject: MgmtTestConfiguration.HasMgmtTestConfiguration(autoRest),
+                LibraryName: GetLibraryNameOption(autoRest, defaultLibraryName),
+                Namespace: GetNamespaceOption(autoRest, defaultNamespace),
+                SkipCSProjPackageReference: GetSkipCSProjPackageReferenceOption(autoRest),
+                RelativeProjectFolder: relativeProjectFolder,
+                Generation1ConvenienceClient: GetGeneration1ConvenienceClientOption(autoRest)
+            );
+        }
+
+        private static string GetOutputFolderOption(IPluginCommunication autoRest) => TrimFileSuffix(GetRequiredOption<string>(autoRest, Options.OutputFolder));
+        private static string? GetProjectFolderOption(IPluginCommunication autoRest) => autoRest.GetValue<string?>(Options.ProjectFolder).GetAwaiter().GetResult();
+        private static bool GetAzureArmOption(IPluginCommunication autoRest) => GetOptionBoolValue(autoRest, Options.AzureArm);
+        private static bool GetSkipCSProjPackageReferenceOption(IPluginCommunication autoRest) => GetOptionBoolValue(autoRest, Options.SkipCSProjPackageReference);
+        private static bool GetGeneration1ConvenienceClientOption(IPluginCommunication autoRest) => GetOptionBoolValue(autoRest, Options.Generation1ConvenienceClient);
+        private static string GetLibraryNameOption(IPluginCommunication autoRest, string defaultLibraryName) => autoRest.GetValue<string?>(Options.LibraryName).GetAwaiter().GetResult() ?? defaultLibraryName;
+        private static string GetNamespaceOption(IPluginCommunication autoRest, string defaultNamespace) => autoRest.GetValue<string?>(Options.Namespace).GetAwaiter().GetResult() ?? defaultNamespace;
     }
+
+    // configuration properties neede by CSharpProj
+    public record CSharpProjConfiguration(
```

Comment:
Shall we separate the `CSharpProjConfiguration` from Configuration. It may not comply c# style, a public struct/class need to be in a .cs file with the same name.

Reply:
done